draft-ietf-masque-connect-udp.md

```diff
@@ -403,18 +428,41 @@ Reference:
 : This document.
 
 
-## Datagram Format Type {#iana-format-type}
+--- back
 
-This document will request IANA to register UDP_PAYLOAD in the "HTTP Datagram
-Format Types" registry established by {{HTTP-DGRAM}}.
+# Example: Registering Contexts with Capsules {#register}
 
-|    Type     |   Value   | Specification |
-|:------------|:----------|:--------------|
-| UDP_PAYLOAD | 0xff6f00  | This Document |
-{: #iana-format-type-table title="Registered Datagram Format Type"}
+Extensions can define a new Capsule type (see {{HTTP-DGRAM}}) that is used to
+register a context ID with the peer endpoint. Registering a context ID is the
+action by which an endpoint informs its peer of the semantics and format of a
+given context ID.
 
+For example, if an extension wanted to convey the time at which a UDP packet was
+received, it could define a new capsule type (REGISTER_TIMESTAMP_CONTEXT_ID)
+that includes a context ID value. Endpoints that understand this new capsule
+type would be able to consequently handle and parse datagrams on the context ID,
+while all other endpoints would ignore the datagrams.
+
+~~~
+REGISTER_TIMESTAMP_CONTEXT_ID Capsule {
```

Comment:
I don't think this is how timestamps or registration should work, so I don't think we should have this example here, at least not in this form.

Reply:
Can you propose an alternative? It's unclear to me why you don't like this example.

Reply:
Proposal [here](https://bemasc.github.io/h3-datagram-ping/bemasc-timestamp-designteam/draft-schwartz-masque-h3-datagram-ping.html#name-timestamp)

Reply:
That's a cool way to do timestamps, but it's too complex for illustration purposes - I'd much rather keep the example in this appendix simple

Reply:
Agreed, but the example shouldn't conflict with any actual proposals.

Reply:
Can you propose another synthetic extension for here that's not timestamps?